### ISSUE TITLE
[Feature] add startServiceSync or transformSync

### ISSUE THREAD

**Issue by aelbore** [evanw/esbuild#136]
- this is useful in some other plugins that doesnt support async/await/promise i think jest doesnt support yet the process async to the transform

**Comment by evanw** [comment]
It's good to know that there's a use case for this. I will consider adding something like this to the official API. Unfortunately it's not possible to benefit from the better performance of the service-oriented API synchronously because it uses streams for communication and stream I/O is asynchronous in node.

It's straightforward to use esbuild in node in a synchronous manner, but you're going to have to incur the overhead of creating a new process every time. Depending on what you're doing, that might be fine. Here's an example of how you might do this, which you can use now without official API support:

```js
function transformSync(input) {
  const execFileSync = require('child_process').execFileSync
  const {buffer} = execFileSync('esbuild', ['--loader=tsx'], {input})
  return Buffer(buffer).toString()
}
```

**Comment by evanw** [comment]
This API is now published in version 0.5.0. The JavaScript API documentation is now [here](https://github.com/evanw/esbuild/blob/master/docs/js-api.md).
